crates/core/src/ring.rs

```diff
@@ -251,62 +381,65 @@ impl Ring {
     ///
     /// # Panic
     /// Will panic if the node checking for this condition has no location assigned.
-    pub fn should_accept(&self, location: &Location) -> bool {
-        let cbl = &*self.connections_by_location.read();
-        let open_conn = self.open_connections.fetch_add(1, SeqCst) + 1;
-        let my_location = &self
+    pub fn should_accept(&self, location: Location) -> bool {
+        let open_conn = self
+            .open_connections
+            .fetch_add(1, std::sync::atomic::Ordering::SeqCst);
+        let my_location = self
             .own_location()
             .location
             .expect("this node has no location assigned!");
-        let accepted = if location == my_location || cbl.contains_key(location) {
+        let accepted = if location == my_location
+            || self.connections_by_location.read().contains_key(&location)
+        {
             false
         } else if open_conn < self.min_connections {
             true
         } else if open_conn >= self.max_connections {
-            tracing::debug!(peer = %self.peer_key, "max connections reached");
+            tracing::debug!(peer = %self.peer_key, "max open connections reached");
             false
         } else {
-            // todo: in the future maybe use the `small worldness` metric to decide
-            let median_distance = self
-                .median_distance_to(my_location)
-                .unwrap_or(Distance(0.5));
-            let dist_to_loc = my_location.distance(location);
-            let is_lower_than_median = dist_to_loc < median_distance;
-            tracing::debug!("dist to connection loc: {dist_to_loc}, median dist: {median_distance}, accepting: {is_lower_than_median}");
-            is_lower_than_median
+            let strategy = if self
+                .fast_acquisition
```

Comment:
May be tricky, I realized that there is no one ideal small world topology, it depends on how big the network is. Small network won't have much clustering, large network will. I think the best metric of how good the topology is is whether outbound requests are evenly distributed among connections - so lower standard deviation is better.